### PR TITLE
Migrate `Tag` to emotion

### PR DESCRIPTION
**Pull request by montezume**

### CODE REVIEW

File: src/components/tag/tag.js [hunk 1]
```diff
@@ -1,51 +1,96 @@
...
+
+const getClickableContentWrapperStyles = ({ type, isRemoveable }) =>
   type === 'warning'
-    ? classnames(styles.clickableContentWrapperWarning, {
-        [styles.clickableRemovableContentWrapperWarning]: isRemovable,
-      })
-    : classnames(styles.clickableContentWrapperNormal, {
-        [styles.clickableRemovableContentWrapperNormal]: isRemovable,
-      });
+    ? [
```
Comment (tdeekens) [inline src/components/tag/tag.js]: Just wondering if we could even do something like:

```js
const getClickableContentWrapperStyles = ({ type, isRemoveable }) => [
  type === 'warning' && css`
    &:hover {
      border-color: ${vars.borderColorTagWarningHover};
      border-right: 1px solid ${vars.borderColorTagPristine};
    }
  `,
  type === 'warning' && isRemoveable && css`
    border-right: 1px solid var(--border-color-tag-warning);
  `

  type =!== 'warning' && css`
    &:hover {
      border-color: ${vars.borderColorTagFocus};
    }
  `
  type =!== 'warning' && isRemoveable && css`
    &:hover {
      border-right: 1px solid ${vars.borderColorTagPristine};
    }
  `
].filter(Boolean)
```

Not sure maybe that's easier to follow. Just what I thought when reading this.

File: src/components/tag/tag.js [hunk 2]
```diff
@@ -1,51 +1,96 @@
...
+        css`
+          &:hover {
+            border-color: ${vars.borderColorTagWarningHover};
+            border-right: 1px solid ${vars.borderColorTagPristine};
+          }
+        `,
+        isRemoveable &&
+          css`
+            border-right: 1px solid var(--border-color-tag-warning);
```
Comment (tdeekens) [inline src/components/tag/tag.js]: Is this a `vars.$` too?

File: src/components/tag/tag.js (hunk 1 above)
Comment (montezume) [reply src/components/tag/tag.js]: I find the whole component hard to follow, I would rather refactor it not to be duplicated with the Link and the Regular, and I find this even more confusing

File: src/components/tag/tag.js (hunk 2 above)
Comment (montezume) [reply src/components/tag/tag.js]: ya thanks 👍

File: src/components/tag/tag.js (hunk 1 above)
Comment (tdeekens) [reply src/components/tag/tag.js]: Sure, it's subjective in the end just wanted to make a suggestion.
